### PR TITLE
Implement UUID type

### LINKED ISSUE #112: Add a UUID type
Create a UUID type in OneType and handle it appropriately in all converters. Only Iceberg has a concept of a UUID and it is a fixed size byte array in the underlying parquet files. We should aim to preserve the metadata that this byte array came from a UUID for round trip conversions.

### PR DESCRIPTION
**Pull request by yyy1000**
## *Important Read*
Closes #112 

## What is the purpose of the pull request

This pull request implements the UUID type in OneType .

## Brief change log

- Add UUID type in OneType
- Handle that in converters

## Verify this pull request
This pull request is already covered by existing tests

### CODE REVIEW

File: core/src/main/java/io/onetable/delta/DeltaSchemaExtractor.java [hunk 1]
```diff
@@ -92,6 +92,7 @@ private DataType convertFieldType(OneField field) {
         return DataTypes.LongType;
       case BYTES:
       case FIXED:
+      case UUID:
```
Comment (yyy1000) [inline core/src/main/java/io/onetable/delta/DeltaSchemaExtractor.java:95]: I think the metadata will be empty for no other logic here, not sure whether this is correct

File: core/src/main/java/io/onetable/avro/AvroSchemaConverter.java [hunk 2]
```diff
@@ -414,6 +414,7 @@ private Schema fromOneSchema(OneSchema oneSchema, String currentPath) {
                       Schema.createFixed(oneSchema.getName(), oneSchema.getComment(), null, size)),
               oneSchema);
         }
+      case UUID:
```
Comment (yyy1000) [inline core/src/main/java/io/onetable/avro/AvroSchemaConverter.java:417]: I don't know how to handle toOneSchema here, for Avro doesn't have a UUID type, the same as DeltaLake.

**Comment by the-other-tim-brown** [comment]
@yyy1000 in the future we want to support "round-trip conversions" where we can convert from Iceberg to Delta and back to Iceberg. One thing to think about is whether we can preserve that context that the byte array is actually meant to represent a UUID in Delta's and Hudi's schema.

File: core/src/main/java/io/onetable/avro/AvroSchemaConverter.java (hunk 2 above)
Comment (the-other-tim-brown) [reply core/src/main/java/io/onetable/avro/AvroSchemaConverter.java:417]: Fixed size byte array makes the most sense in my opinion

**Comment by yyy1000** [comment]
Thx for your guidance! I will try to implement that.
A question is how to handle BYTES in toOneSchema function, whether it should be converted to UUID or BYTES?

**Comment by the-other-tim-brown** [comment]
> Thx for your guidance! I will try to implement that. A question is how to handle BYTES in toOneSchema function, whether it should be converted to UUID or BYTES?

For which format?

**Comment by yyy1000** [comment]
> > Thx for your guidance! I will try to implement that. A question is how to handle BYTES in toOneSchema function, whether it should be converted to UUID or BYTES?
> 
> For which format?

Both Avro and Delta. 
e.g., the BYTES type in Avro schema can possibly be from OneType.Bytes or OneType.UUID.

**Comment by the-other-tim-brown** [comment]
> > > Thx for your guidance! I will try to implement that. A question is how to handle BYTES in toOneSchema function, whether it should be converted to UUID or BYTES?
> > 
> > 
> > For which format?
> 
> Both Avro and Delta. e.g., the BYTES type in Avro schema can possibly be from OneType.Bytes or OneType.UUID.

@yyy1000 apologies for the delay. I was thinking that we would need to track some metadata within these schema fields for Avro and Delta. If there is no metadata indicating it's a special case, it falls back to a regular Bytes case. Let me know if you have any other ideas as well for how we can carry that context through.

**Comment by gzagarwal** [comment]
@yyy1000 @the-other-tim-brown  is this PR completed?

**Comment by yyy1000** [comment]
No @gzagarwal , I think you can take in the original issue.

**Pull request closed without merging**